src/Symfony/Component/Amqp/Exchange.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Amqp;
+
+use Symfony\Component\Amqp\Exception\LogicException;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class Exchange extends \AMQPExchange
+{
+    /**
+     * Create a new Exchange.
+     *
+     * Special arguments:
+     *  * flags: if set, setFlags() will be called with its value
+     *  * type: the queue type (set by setType())
+     */
+    public function __construct(\AMQPChannel $channel, string $name, array $arguments = array())
+    {
+        parent::__construct($channel);
+
+        parent::setName($name);
+
+        if (Broker::DEAD_LETTER_EXCHANGE === $name) {
```

Comment:
Outch. That's super hardcoding: can you move this logic (i.e. name to type mapping) to wherever this is created?